lib/improver/utilities/temporal_interpolation.py

```diff
@@ -201,8 +200,16 @@ def check_cube_coords_dtype(cube, new_cube, dim_coords=None):
             dtype_orig = cube.coord(coord_name).points.dtype
             try:
                 new_coord = new_cube.coord(coord_name)
-                check_coord_datatypes(new_coord, dtype_orig, fix=True,
-                                      rounding=True)
+                # TODO: Provide safe support for time coordinates that could
+                # potentially be in units of "hours since 1970-01-01 00:00:00."
+                if new_coord.points.dtype != dtype_orig:
+                    new_coord.points = np.around(new_coord.points)
+                    new_coord.points = new_coord.points.astype(dtype_orig)
```

Comment:
This still isn't safe.  Please could we add either a unit conversion / check, or an error that is thrown if the coordinate values are changed at anything more than float precision level.

Reply:
If you want to change the datatype, you HAVE to change the units.   You can't convert the units back into hours.  That retains the risk of truncating sub-hourly times.

Reply:
Thanks @cgsandford, I've updated now removed the unnecessary unit conversion: https://github.com/metoppv/improver/pull/808/commits/fe835e870c4efdf24f0d3c7b71cad040dac9b0ae